### PR TITLE
test(HttpAppClient): Fail if response is written twice

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

File: zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala [hunk 1]
```diff
@@ -100,6 +100,7 @@ object HttpAppClient {
     allowedThread: Thread,
   ) extends EmbeddedChannel() { self =>
     private var pendingRead: Boolean = false
+    private var canWrite             = true
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala]: add scala doc and use PendingWrite variable and reverse the logic instead

File: zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala [hunk 2]
```diff
@@ -148,11 +149,22 @@ object HttpAppClient {
...
+        if (canWrite) {
+          ZIO.effect(if (!msg.isInstanceOf[LastHttpContent]) {
+            canWrite = false
+
+          } else {
+            canWrite = true
+          }) *> outbound.offer(msg.asInstanceOf[HttpObject])
+
+        } else {
+          ZIO.fail(new Error("Unable to write"))
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala]: Give specific and detailed error

File: zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala [hunk 3]
```diff
@@ -148,11 +149,22 @@ object HttpAppClient {
      */
     override def handleOutboundMessage(msg: AnyRef): Unit = {
       assertThread("handleOutboundMessage")
-      rtm
-        .unsafeRunAsync(outbound.offer(msg.asInstanceOf[HttpObject])) {
-          case Exit.Failure(cause) => System.err.println(cause.prettyPrint)
-          case _                   => ()
+      rtm.unsafeRunAsync {
+        if (canWrite) {
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala]: move this condition outside of unsaferun

**Comment by tusharmath** [comment]
not required

**Pull request closed without merging**